packages/playground/data-liberation/src/import/WP_Topological_Sorter.php

```diff
@@ -70,10 +59,58 @@ class WP_Topological_Sorter {
 	 */
 	protected $current_item = 0;
 
+	const ENTITY_TYPES = array(
+		'comment'      => 1,
+		'comment_meta' => 2,
+		'post'         => 3,
+		'post_meta'    => 4,
+		'term'         => 5,
+	);
+
+	private $mapped_pre_filters = array(
+		// Name of the filter, and the number of arguments it accepts.
+		'wxr_importer_pre_process_comment' => 2,
+		'wxr_importer_pre_process_comment_meta' => 2,
+		'wxr_importer_pre_process_post' => 2,
+		'wxr_importer_pre_process_post_meta' => 2,
+		'wxr_importer_pre_process_term' => 1,
+	);
+
+	private $mapped_post_actions = array(
+		// Name of the filter, and the number of arguments it accepts.
+		'wxr_importer_processed_comment' => 3,
+		'wxr_importer_processed_comment_meta' => 3,
+		'wxr_importer_processed_post' => 2,
+		'wxr_importer_processed_post_meta' => 2,
+		'wxr_importer_processed_term' => 2,
+	);
+
 	public function __construct( $options = array() ) {
 		if ( array_key_exists( 'session_id', $options ) ) {
 			$this->current_session = $options['session_id'];
 		}
+
+		// The topological sorter needs to know about the mapped IDs for comments, terms, and posts.
+		foreach ( $this->mapped_pre_filters as $name => $accepted_args ) {
+			add_filter( $name, array( $this, 'filter_wxr_importer_pre_process' ), 10, $accepted_args );
+		}
+
+		foreach ( $this->mapped_post_actions as $name => $accepted_args ) {
+			add_action( $name, array( $this, 'action_wxr_importer_processed' ), 10, $accepted_args );
+		}
+	}
+
+	/**
+	 * Remove the filters.
+	 */
+	public function __destruct() {
```

Comment:
I'm always nervous about coupling cleanup logic with GC behavior we don't control. Will `__destruct` be called while `$this` reference is present in the list of filters? I don't know, but I'd guess "no". With an explicit `cleanup()` or so method, we'd never have to guess.

Reply:
Make sense, I will add the unregistering to the deactivate method.